examples/dmm/polyphonic_data_loader.py

```diff
@@ -44,13 +44,20 @@ def process_data(base_path, filename, T_max=160, min_note=21, note_range=88):
                 slice_length = len(note_slice)
                 if slice_length > 0:
                     processed_dataset[split]['sequences'][seq, t, note_slice] = np.ones((slice_length))
-    pickle.dump(processed_dataset, open(output, "wb"))
+    pickle.dump(processed_dataset, open(output, "wb"), pickle.HIGHEST_PROTOCOL)
     print("dumped processed data to %s" % output)
 
 
 # this logic will be initiated upon import
 base_path = './data'
 process_data(base_path, "jsb_processed.pkl")
+jsb_file_loc = "./data/jsb_processed.pkl"
+
+
+# ingest training/validation/test data from disk
+def load_data():
+    with open(jsb_file_loc, "rb") as f:
+        return pickle.load(f)
```

Comment:
Pickling / unpickling is incompatible between python 2 and 3 due to ambiguity in bytestring decoding ([stackoverflow issue](https://stackoverflow.com/questions/28218466/unpickling-a-python-2-object-with-python-3)). We can include the `encoding='latin-1'` for python 3 to ensure compatibility. Another thing would be to force remove the dataset and download again if you change your virtual env.

Reply:
Good point. I'll simply remove the file if it is incompatible. `HIGHEST_PROTOCOL` is much faster, and that's the most common use case.

Reply:
Sounds good!